spec/inputs/beats_spec.rb

```diff
@@ -117,6 +118,44 @@
           expect(event).to be_nil
         end
       end
+
+      context "with a beat.hostname field" do
+        let(:event_map) { {"message" => "hello", "beat" => {"hostname" => "linux01"} } }
+
+        it "copies it to the host field" do
+          event = beats.create_event(event_map, identity_stream)
+          expect(event["host"]).to eq("linux01")
+        end
+      end
+
+      context "without a beat.hostname field" do
+        let(:event_map) { {"message" => "hello", "beat" => {"name" => "linux01"} } }
+
+        it "should not add a host field" do
+          event = beats.create_event(event_map, identity_stream)
+          expect(event["beat"]["name"]).to eq("linux01")
+          expect(event["host"]).to be_nil
+        end
+      end
+
+      context "with a beat.hostname and host fields" do
+        let(:event_map) { {"message" => "hello", "host" => "linux02", "beat" => {"hostname" => "linux01"} } }
+
+        it "should not overwrite host" do
+          event = beats.create_event(event_map, identity_stream)
+          expect(event["host"]).to eq("linux02")
+        end
+      end
+
+      context "with a host field in the message" do
+        let(:codec) { LogStash::Codecs::JSON.new }
+        let(:event_map) { {"message" => '{"host": "linux02"}', "beat" => {"hostname" => "linux01"} } }
+
+        it "should take the host from the JSON message" do
+          event = beats.create_event(event_map, identity_stream)
+          expect(event["host"]).to eq("linux02")
+        end
```

Comment:
There is only small problem with the tests. We use the `message` field to check if we should apply a codec or not to the event and theses tests will only trigger the `else` part of this conditionals. https://github.com/logstash-plugins/logstash-input-beats/pull/28/files#diff-28d8b79a78399c1304c58e81cd3672f2R140